includes/Product/functions.php

```diff
@@ -543,6 +543,10 @@ function dokan_get_vendor_by_product( $product, $id = false ) {
 
     $vendor_id = apply_filters( 'dokan_get_vendor_by_product', $vendor_id, $product );
 
+    if ( ! $vendor_id ) {
+        return false;
+    }
+
     return false === $id ? dokan()->vendor->get( $vendor_id ) : (int) $vendor_id;
```

Comment:
instead of returning false, what you can do is on line 550
```
return false === $id  && ! empty( $vendor_id ) ? dokan()->vendor->get( $vendor_id ) : (int) $vendor_id;
```

Reply:
also you can change the parameter $id  with  `$get_id` and change accordingly